.github/workflows/README.md

```diff
@@ -0,0 +1,115 @@
+# Workflow Taxonomy
+
+## Where and what tests are run
```

Comment:
Is this document straight forward on how to skim? I tried to drill down as far as it made sense to and expose what was actually being called by these workflows.

Reply:
it's pretty straight forward to skim 🎉  Thanks for doing this 😄 

I guess my only concern is that if any of the code implementation changes, we'd need to update the docs as well, so we'd sort of have two sources of truth. But I don't think this part of code changes often, so probably not a huge concern?

Reply:
I have a similar concern; while nice to have the code snippets near the docs, I actually think it'd be easier to maintain and similarly clear if we replaced code snippets with descriptions of what each type of test is meant to accomplish and whether it's run in a deployed env vs. a container

Reply:
Do these GH code snippets work in GH markdown? Is that a good middle ground, if so?